nixos/lib/test-driver/test-driver.py

```diff
@@ -0,0 +1,680 @@
+#! /somewhere/python3
+
+from __future__ import print_function
+from xml.sax.saxutils import XMLGenerator
```

Comment:
Can we split some of the classes into separate files? I was a bit baffled about why we need to generate xml, and the list is quite long.

Reply:
I think we can drop the XML log file stuff. This was for generating pretty-printed logs with collapsible structure, but I don't think anybody cares about keeping that.

Reply:
@edolstra I think that the xml/html log help to give context to the log entries and to tie them to the imperative command that is the cause for their generation. It does a very good job at it, with its tree-like rendering.  The only thing that i find strange is the face that when the test fails, it isn't generated. It would be much more useful to understand what happened if we could have it when the test fails

Reply:
That was done on purpose to unblock issues with Hydra IIRC.